scapy/layers/dns.py

```diff
@@ -1432,18 +1447,22 @@ def parse_options(self, joker=None,
                       joker6=False,
                       send_error=False,
                       relay=False,
-                      from_ip=None,
-                      from_ip6=None,
+                      from_ip=True,
+                      from_ip6=False,
                       src_ip=None,
                       src_ip6=None,
-                      ttl=10,
-                      jokerarpa=None):
+                      ttl=60,
+                      jokerarpa=False):
         """
-        :param joker: default IPv4 for unresolved domains. (Default: None)
+        Simple DNS answering machine.
+
+        :param joker: default IPv4 for unresolved domains.
                       Set to False to disable, None to mirror the interface's IP.
+                      Defaults to None, unless 'match' is used, then it defaults to
+                      False.
         :param joker6: default IPv6 for unresolved domains (Default: False)
                        set to False to disable, None to mirror the interface's IPv6.
```

Comment:
(It's unrelated to this PR but `joker6=None` doesn't seem to be fully compatible with mDNS. It doesn't extract link-local addresses so no replies are sent on machines with no global addresses). According to https://datatracker.ietf.org/doc/html/rfc6762#section-6.2 all the valid addresses should be sent:
>    When a Multicast DNS responder sends a Multicast DNS response message
   containing its own address records, it MUST include all addresses
   that are valid on the interface on which it is sending the message,
   and MUST NOT include addresses that are not valid on that interface
   (such as addresses that may be configured on the host's other
   interfaces)